cudawrapper.c

```diff
@@ -372,14 +372,23 @@ gpu_ecm (mpz_t f, mpz_t x, int param, mpz_t firstsigma, mpz_t n, mpz_t go,
                               TreeFilename, maxmem, Fermat, modulus);
   if (youpi == ECM_ERROR)
       goto end_gpu_ecm;
-  
+
+  int schedule = 0;
+  if (use_cgbn)
+    {
+#ifdef HAVE_CGBN_H
+    schedule = 1;
+#else 
+    outputf (OUTPUT_ERROR, "cgbn not included, using yield schedule");
```

Comment:
This will error on line 482.

Maybe

int schedule = (use_cgbn & HAVE_CGBN_H) ? 1 : 0

Reply:
I've left the error on line 482 alone.  If line 482 happens, then the program exits.  This is expected in your original code, right?  I think the only we need to change is to remove the "#else" here.  No need to let the user know about this issue if the program will quit at line 482 later.

Also, your proposed solution would probably fail to compile if HAVE_CGBN_H isn't defined.

I'll remove the "#else".

Reply:
how about 

`int schedule = use_cgbn ? 0 : 1`

technically if you call ecm with `-gpu -cgbn` and cgbn wasn't configured the card will be initialized with the wrong scheduler but it's going to exit 0.001 seconds later.

Reply:
Sure, that sounds good.  It looks cleaner too.  I'll update that.

Reply:
I've updated that line.  You can see the commit down below.